extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/security/AbstractPermissionsAllowedTestCase.java

```diff
@@ -0,0 +1,129 @@
+package io.quarkus.resteasy.reactive.server.test.security;
+
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.security.StringPermission;
+import io.quarkus.security.test.utils.TestIdentityController;
+import io.restassured.RestAssured;
+
+public abstract class AbstractPermissionsAllowedTestCase {
+
+    @BeforeAll
+    public static void setupUsers() {
+        TestIdentityController.resetRoles()
```

Comment:
I think the permission checkers should be added the way users can add them, via a custom `SecurityIdentityAugmentor`

Reply:
In that case I can't mock identity, so I'll have to set up security for all of these tests. Also all these `assertSuccess` must be abandoned and rewrote. These are just junit tests. There is integration tests and RESTEasy Reactive tests that use actual identity (and testing identity respectively).

Reply:
I could trick it somehow, but this tests principle (no matter of how you get permission, your SecurityIdentity has it..). Is it important?

Reply:
Sorry, now I mentioned this is a comment on RESTEasy Reactive, I was worried you want me to change tests in security module. I'll use augmentor here, no problem.

Reply:
Done